File: src/main/java/p00/Problem_0003.java
package p00;

public class Problem_0003 {

	public static int solve() {

		long number = 600851475143L;
		int prime = 2;

		while (number > 1) {

			if (number % prime == 0)
				number /= prime;
			else
				prime++;

		}
		return prime;
	}

}

File: src/main/csharp/p00/Problem0003.cs
namespace ProjectEuler.main.csharp.p00
{
    public class Problem0003
    {
        public static long Solve()
        {
            var num = 600851475143;

            for (var i = 2; i < num - 1; i++)
            {
                if (num % i == 0)
                {
                    num /= i;
                    i--;
                }
            }

            return num;
        }
    }
}